public/app/features/logs/components/LogRows.tsx

```diff
@@ -112,6 +112,18 @@ class UnThemedLogRows extends PureComponent<Props, State> {
     sortLogRows(logRows, logsSortOrder)
   );
 
+  /**
+   * This method is used to prevent unnecessary re-renders of the component.
+   * If the log rows don't change, the component won't re-render
+   * @param nextProps
+   */
```

Comment:
i dont think we need to repeat React's docs about what `shouldComponentUpdate()` is for :rofl: